### ISSUE TITLE
Incorrect bit packing / unpacking algorithm with small pallet size

### ISSUE THREAD

**Issue by KokeCacao** [misode/vscode-nbt#76]
Not sure whether it is intended, but litematica uses a different bit packing strategy for small pallet size. Your implementation looks different from either Minecraft or litematica.

Here is a simple python code on how to correct this.

```
import math
MINECRAFT_BITS_PER_ENTRY_FN = lambda pallette_size: max(4, math.ceil(math.log2(pallette_size)))
LITEMATICA_BITS_PER_ENTRY_FN = lambda pallette_size: max(2, (pallette_size - 1).bit_length())
YOUR_IMPLEMENTATION = lambda pallette_size: max(1, (pallette_size - 1).bit_length())
```

I also created a test file that should display a ocean chunk in litematica. But your 3D viewer display slices.
[0_4.log](https://github.com/user-attachments/files/17807636/0_4.log)
(change the .log extension to .litematica)

**Comment by misode** [comment]
Thanks for the thorough explanantion and reproduction file!

**Comment by KokeCacao** [comment]
Thanks for your reply. I kinda need the fix ASAP so I did it: https://github.com/misode/vscode-nbt/pull/77

**Comment by misode** [comment]
Thanks for the fix. I've published a new version v0.9.3
